eclair-core/src/test/scala/fr/acinq/eclair/channel/states/g/NegotiatingStateSpec.scala

```diff
@@ -130,11 +124,11 @@ class NegotiatingStateSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike
     assert(alice.stateData.asInstanceOf[DATA_NEGOTIATING].closingTxProposed.length == 1)
     assert(alice.stateData.asInstanceOf[DATA_NEGOTIATING].closingTxProposed.last.length == 1)
     assert(alice.stateData.asInstanceOf[DATA_NEGOTIATING].bestUnpublishedClosingTx_opt.isEmpty)
-    if (alice.stateData.asInstanceOf[DATA_NEGOTIATING].commitments.channelFeatures.hasFeature(Features.UpfrontShutdownScript)) {
+    if (alice.stateData.asInstanceOf[DATA_NEGOTIATING].commitments.localParams.upfrontShutdownScript_opt.isDefined) {
       // check that the closing tx uses Alice and Bob's default closing scripts
       val closingTx = alice.stateData.asInstanceOf[DATA_NEGOTIATING].closingTxProposed.last.head.unsignedTx.tx
-      val expectedLocalScript = alice.stateData.asInstanceOf[DATA_NEGOTIATING].commitments.localParams.defaultFinalScriptPubKey
-      val expectedRemoteScript = bob.stateData.asInstanceOf[DATA_NEGOTIATING].commitments.localParams.defaultFinalScriptPubKey
+      val expectedLocalScript = alice.stateData.asInstanceOf[DATA_NEGOTIATING].localShutdown.scriptPubKey
+      val expectedRemoteScript = bob.stateData.asInstanceOf[DATA_NEGOTIATING].localShutdown.scriptPubKey
```

Comment:
This is wrong on two accounts:
- the `if` should have kept looking at `channelFeatures.hasFeature(Features.UpfrontShutdownScript)`
- `expectedLocalScript` should be be taken from `localParams.upfrontShutdownScript_opt.get` (same for remote)

Reply:
done in done in https://github.com/ACINQ/eclair/pull/2565/commits/a3da92a3f0bac4f0d093ce5548eb89f5b1287ed1